SDK v2 migration guide.md

```diff
@@ -16,182 +16,189 @@ are outlines of the notable breaking changes as well as a mapping from version 1
+  > Be advised, subscriptions do not carry over when the client is re-opened.
+  >
+  > It cannot be re-opened after disposal.
+
+#### API mapping
+
+| v1 API | Equivalent v2 API | Notes |
+|:---|:---|:---|
+| `DeviceClient` | `IotHubDeviceClient` | Specify the service it is a device client for. |
+| `DeviceClient.SendEventAsync(...)` | `IotHubDeviceClient.SendTelemetryAsync(...)` | Even our public documentation calls this telemetry, so we renamed the method to describe this better.¹ |
+| `DeviceClient.SendEventBatchAsync(...)` | `IotHubDeviceClient.SendTelemetryBatchAsync(...)` | See¹ |
+| `DeviceClient.SetConnectionStatusChangesHandler(...)` | `IotHubDeviceClient.ConnectionStatusChangeCallback` | Local operation doesn't require being a method. |
+| `DeviceClient.SetReceiveMessageHandlerAsync(...)` | `IotHubDeviceClient.SetIncomingMessageCallbackAsync(...)` | Disambiguate from telemetry messages. |
+| `DeviceClient.GetTwinAsync(...)` | `IotHubDeviceClient.GetTwinPropertiesAsync(...)` | The device client doesn't get the full twin, just the properties so this helps avoid that confusion.² |
+| `Twin` | `TwinProperties` | See² |
+| `Twin.Properties.Desired` | `TwinProperties.Desired` | See² |
+| `Twin.Properties.Reported` | `TwinProperties.Reported` | See² |
+| `MessageResponse` | `MessageAcknowledgement` | It isn't a full response, just a simple acknowledgement. |
+| `Message` | `TelemetryMessage`, `IncomingMessage` | Distinguished between the different messaging operations. |
+| `DeviceClient.SetRetryPolicy(...)` | `IotHubClientOptions.RetryPolicy` | Should be specified at initialization time, and putting it in the client options object reduces the client API surface. |
+| `ExponentialBackOff` | `ExponentialBackOffRetryPolicy` | Clarify it is a retry policy. |
+| `Message.CreationTimeUtc` | `TelemetryMessage.CreatedOnUtc`, `IncomingMessage.CreatedOnUtc` | Conforming to the naming guidelines by the Azure SDK team, where DateTime/Offset types have an "On" suffix (and "Utc" suffix when explicitly in UTC).³ |
+| `Message.EnqueuedTimeUtc` | `TelemetryMessage.EnqueuedtimeUtc`, `IncomingMessage.EnqueuedTimeUtc` | See³ |
+| `Message.ExpiryTimeUtc` | `TelemetryMessage.ExpiresOnUtc`, `IncomingMessage.ExpiresOnUtc` | See³ |
+| `MethodRequest` | `DirectMethodRequest` | Use full name of the operation type.⁴ |
+| `MethodResponse` | `DirectMethodResponse` | See⁴ |
 
 #### ModuleClient
 
-| Version 1 API | Equivalent version 2 API |
-|:---|:---|
-| `ModuleClient` | `IotHubModuleClient` |
-| `MessageResponse` | `MessageAcknowledgement` |
-| `Message` | `IncomingMessage`, `OutgoingMessage` |
+The device client and module client share a lot of API surface and underlying implementation. See changes to the IoT hub [device client](#deviceclient).
 
-#### Other notable breaking changes
+#### Notable breaking changes
 
-- See changes to `DeviceClient`.
```

Comment:
Why we are stating "N/A" instead of this?

Reply:
See line 177 in the ne wfile.

Reply:
My bad, I missed it.

Reply:
How about we remove this header altogether? Someone could potentially skip the previous statement and think there were no breaking changes.

Reply:
Or, "see above"?